doc/configuration.rst

```diff
@@ -18,12 +18,94 @@ The name can be seem from ``doit help`` output::
    -f ARG, --file=ARG        load task from dodo FILE [default: dodo.py]  (config: dodoFile)
 
 
+pyproject.toml
+--------------
+
+`doit` configuration can be read from `pyproject.toml <https://www.python.org/dev/peps/pep-0518/>`_
+under the `tool.doit` namespace. In the future, especially if a TOML parser becomes
+part of the python standard library, this will become the preferred configuration source,
+and may gain features not available in the legacy `doit.cfg`.
+
+.. note::
+
+   As a TOML parser is _not_ yet part of the standard library, a third-party package is
+   required, one of:
+
+   - `toml <https://pypi.org/project/toml/>`_
+   - `tomlkit <https://pypi.org/project/tomlkit/>`_
+
+
+TOML vs INI
+^^^^^^^^^^^
+
+While mostly similar, `TOML <https://toml.io>`_ differs from the INI format
+in a few ways:
+
+- all strings must be quoted with `'` or `"`
+- triple-quoted strings may contain new line characters (`\n`) and quotes
+- must be saved as UTF-8
+- integers and floating point numbers can be written without quotes
+- boolean values can be written unquoted and lower-cased, as `true` and `false`
+
+Unlike "plain" TOML, `doit` will parse pythonic strings into their correct types,
```

Comment:
Why? I think this is in anti-feature. And I could not see where this comes from, all python libs work like this?

Reply:
This is a hangover from being able to reuse the existing `.ini` parsing logic, which only speaks string.

Reply:
oh. i see... I think that is not acceptable.

Reply:
Welp, that `config` object (wherever it gets read from or built dynamically) gets handed around throughout the entire running application, it appears, and (only) the `cmd_options` know how to do the marshaling. It seems like changing (or even detecting the use of) that behavior would require deprecation of the ini format altogether, and a formal data schema throughout all these things... sounds very 1.0ish.